components/org.wso2.carbon.appmgt.mdm.osgiconnector/src/main/java/org/wso2/carbon/appmgt/mdm/osgiconnector/ApplicationOperationsImpl.java

```diff
@@ -0,0 +1,251 @@
+			DeviceIdentifier deviceIdentifier;
+			for (String param : applicationOperationAction.getParams()) {
+				deviceIdentifier = new DeviceIdentifier();
+				String[] paramDevices = param.split("---");
+				deviceIdentifier.setId(paramDevices[0]);
+				deviceIdentifier.setType(paramDevices[1]);
+				deviceIdentifiers.add(deviceIdentifier);
+			}
+		} else {
+			throw new IllegalStateException("invalid type is received from app store.");
+		}
+		App app = applicationOperationAction.getApp();
+		MobileApp mobileApp = new MobileApp();
+		mobileApp.setId(app.getId());
+		mobileApp.setType(MobileAppTypes.valueOf(app.getType().toUpperCase()));
+		mobileApp.setAppIdentifier(app.getAppIdentifier());
+		mobileApp.setIconImage(app.getIconImage());
+		mobileApp.setIdentifier(app.getIdentifier());
+		mobileApp.setLocation(app.getLocation());
+		mobileApp.setName(app.getName());
+		mobileApp.setPackageName(app.getPackageName());
+		mobileApp.setPlatform(app.getPlatform());
+		mobileApp.setVersion(app.getVersion());
+		Properties properties = new Properties();
+
+		if (MDMAppConstants.IOS.equals(app.getPlatform())) {
+			if (MDMAppConstants.ENTERPRISE.equals(app.getType())) {
+				properties.put(MDMAppConstants.IOSConstants.IS_REMOVE_APP, true);
+				properties.put(MDMAppConstants.IOSConstants.IS_PREVENT_BACKUP, true);
+			} else if (MDMAppConstants.IOSConstants.PUBLIC.equals(app.getType())) {
+				properties.put(MDMAppConstants.IOSConstants.I_TUNES_ID,
+				               Integer.parseInt(app.getIdentifier()));
+				properties.put(MDMAppConstants.IOSConstants.IS_REMOVE_APP, true);
+				properties.put(MDMAppConstants.IOSConstants.IS_PREVENT_BACKUP, true);
+			} else if (MDMAppConstants.WEBAPP.equals(app.getType())) {
+				properties.put(MDMAppConstants.IOSConstants.LABEL, app.getName());
+				properties.put(MDMAppConstants.IOSConstants.IS_REMOVE_APP, true);
+			}
+		} else if (MDMAppConstants.WEBAPP.equals(app.getPlatform())) {
+			properties.put(MDMAppConstants.IOSConstants.LABEL, app.getName());
+			properties.put(MDMAppConstants.IOSConstants.IS_REMOVE_APP, true);
+		}
+		mobileApp.setProperties(properties);
+		try {
+			for (DeviceIdentifier deviceIdentifier : deviceIdentifiers) {
+				if (deviceIdentifier.getType().equals(Platform.android.toString())) {
+					if (MDMAppConstants.INSTALL.equals(applicationOperationAction.getAction())) {
+						operation = AndroidApplicationOperationUtil
+								.createInstallAppOperation(mobileApp);
+					} else {
+						operation = AndroidApplicationOperationUtil
+								.createAppUninstallOperation(mobileApp);
+					}
+				} else if (deviceIdentifier.getType().equals(Platform.ios.toString())) {
+					if (MDMAppConstants.INSTALL.equals(applicationOperationAction.getAction())) {
+						operation =
+								IOSApplicationOperationUtil.createInstallAppOperation(mobileApp);
+					} else {
+						operation =
+								IOSApplicationOperationUtil.createAppUninstallOperation(mobileApp);
+					}
+				}
```

Comment:
What happens for the `Platform.winows`? Isn't it relevant in here?
https://github.com/wso2/carbon-device-mgt/blob/e537e2d841ca3c7946cd34e9da96f261e0ac94e1/components/device-mgt/org.wso2.carbon.device.mgt.common/src/main/java/org/wso2/carbon/device/mgt/common/Platform.java